### PR TITLE
Add `entities.getCount({query})` API.

### PR DESCRIPTION
**Pull request by JSAssassin**

### CODE REVIEW

**Review by mandyvenables — APPROVED** [review]

File: lib/entities.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
     collection: 'tokenization-entity',
     fields: {'entity.internalId': 1},
     options: {unique: true, background: false}
+  }, {
+    collection: 'tokenization-entity',
+    fields: {'meta.created': 1}
```
Comment (dlongley) [inline lib/entities.js:21]:
```suggestion
    fields: {'meta.created': 1},
    options: {unique: false, background: false}
```

File: lib/entities.js [hunk 2]
```diff
@@ -477,3 +480,17 @@ export async function _upsert({
...
  * @typedef {object} ExplainObject
  */
+
+/**
+ * Retrieves the total count of entities matching the given query.
+ *
+ * @param {object} options - The options to use.
+ * @param {object} [options.query={}] - The query to use.
+ *
+ * @returns {Promise<number>} Resolves to the total count of entities that
```
Comment (dlongley) [inline lib/entities.js:490]: Update based on comment below.

File: test/mocha/mock.data.js [hunk 3]
```diff
@@ -223,7 +223,7 @@ export const mockEntity2 = {
     // internalId is generated in tests.
     batchInvalidationCount: 0,
     openBatch: {},
-    minAssuranceForResolution: 2,
+    minAssuranceForResolution: 1,
```
Comment (dlongley) [inline test/mocha/mock.data.js:226]: This is used in existing tests ... I'd rather us not change it but add another entity if we need another one.

**Review by dlongley — CHANGES_REQUESTED** [review]
Thanks! A few suggestions here and a conflict to resolve (since the other PR was merged).

File: lib/entities.js (hunk 1 above)
Comment (JSAssassin) [reply lib/entities.js]: Specified options [75d5cec](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/75d5cecb8ea0e12d11ec0f10a1a350d2ec736864)

File: lib/entities.js [hunk 4]
```diff
@@ -477,3 +480,17 @@ export async function _upsert({
  *
  * @typedef {object} ExplainObject
  */
```
[the remark this answers is not in the document]
Comment (JSAssassin) [reply lib/entities.js]: Moved [1c4d824](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/1c4d8242ea70a5cf59b6077a89fa0d949966bfc4)

File: lib/entities.js [hunk 5]
```diff
@@ -477,3 +480,17 @@ export async function _upsert({
...
+ *
+ * @param {object} options - The options to use.
+ * @param {object} [options.query={}] - The query to use.
+ *
+ * @returns {Promise<number>} Resolves to the total count of entities that
+ *   matched the query.
+ */
+export async function getCount({query = {}} = {}) {
+  const collection = database.collections['tokenization-entity'];
+  return collection.countDocuments(query);
```
[the remark this answers is not in the document]
Comment (JSAssassin) [reply lib/entities.js]: Thanks, updated [f32f90b](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/f32f90bedf0957d721b50995335e924cf51108af)

File: lib/entities.js (hunk 2 above)
Comment (JSAssassin) [reply lib/entities.js]: Updated [f32f90b](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/f32f90bedf0957d721b50995335e924cf51108af)

File: test/mocha/mock.data.js (hunk 3 above)
Comment (JSAssassin) [reply test/mocha/mock.data.js:240]: Added another mock entity [14ea658](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/14ea65865f175bbc56fc4dcf806a01e0e266ab13)

**Comment by mattcollier** [comment]
@dlongley we know we have a use case that involves `minAssuranceForResolution`, should there be an index on that field?

File: lib/entities.js [hunk 6]
```diff
@@ -16,6 +16,14 @@ bedrock.events.on('bedrock-mongodb.ready', async () => {
     collection: 'tokenization-entity',
     fields: {'entity.internalId': 1},
     options: {unique: true, background: false}
+  }, {
+    collection: 'tokenization-entity',
+    fields: {'meta.created': 1},
+    options: {unique: false, background: false}
+  }, {
+    collection: 'tokenization-entity',
+    fields: {'entity.minAssuranceForResolution': 1},
```
Comment (dlongley) [inline lib/entities.js:25]:
```suggestion
    fields: {'entity.minAssuranceForResolution': 1, 'meta.created': 1},
```

Comment (JSAssassin) [reply lib/entities.js]: Updated [9d44df9](https://github.com/digitalbazaar/bedrock-tokenization/pull/106/commits/9d44df9a0beed9b7809b185ce584ab77d0d19628)

**Pull request merged**